README.md

```diff
@@ -70,3 +70,7 @@ Performance optimizations are up to implementations and are not guaranteed by th
 ### Are consumers going to do anything other than convert when they encounter ill-formed strings? If so, why not provide only a conversion method with a fast path for well-formed strings?
 
 Consumers may want to throw/error when encountering ill-formed strings. Also, consumers may want to defer the conversion or the error until later when the String is actually interpreted as Unicode text. These use cases justify the test-only method.
+
+### How does this proposal relate to the WebAssembly Component Model?
```

Comment:
This PR seems fine to me now, if this entire superfluous section is removed.

Reply:
There are several direct connections to this proposal in this section of the FAQ (that I moved there from the problem statement to do ya'll a favor btw). In addition to explaining why the CM is probably the most notable precedent for this proposal, why timings overlap, why this is special, why performance is important, etc., the paragraph also aids future discussion. Today, this proposal uses the CM as an argument, and it is likely that tomorrow the CM will use this proposal as an argument to bolster its choices because "but you can check". This is highly relevant context that, if withheld for editorial reasons, will not be obvious and likely lead to worse results overall than if it was provided. Please let's be responsible.

Reply:
As a JS developer, I should not need to understand or read anything about wasm beyond "wasm requires well-formed strings". None of the info about the component model is relevant to me. A TC39 proposal is not an appropriate place to create leverage for an argument within wasm.

Reply:
This is not intended as leverage, really. If you can point me to sentences that feel like it to you, please tell me, and I'll do my best to phrase as objectively as I humanly can. Hardly anyone knows about the nitty gritty details in play here (this proposal helps to some extent), even less people know how all this is connected, but exactly this knowledge is important, now more than ever with the CM doing things differently than the platform has done it before. As a JS developer, you should care because you have dependencies, and these dependencies have dependencies, you upgrade them, you bundle them. This string stuff is subtle, but also fundamental, and errors are delayed, almost impossible to spot. With the CM, JS developers need a bunch of highly specialized knowledge now that they didn't necessarily need to care about before. I do think that we are acting responsible when we try our best to let them know about the what and why, so they can address it, either in code or in the spec. Please give me a realistic chance to improve this FAQ section. I think it's extremely important that people are aware.

Reply:
I think the place that awareness should be spread is a wasm arena, not a JS one.

Reply:
Please, this is about helping JS devs. To help TC39. ECMAScript. I am out of options in Wasm, nobody cares there :(

Reply:
That's exactly the point - if wasm's decisions here aren't going to change, why is this extra context useful to JS devs? All they need to know is _what_ a well-formed string is, how to check for it, and how to sanitize one if needed - which this proposal already provides for.

Reply:
Given that the CM is in phase 1, it's too early to know for sure whether Wasm's decisions are going to last. And as I said in my prior comment, JS devs, in particular those intending to consume Wasm modules, even more so those with an interest to use JS and Wasm in tandem (which is a fantastic use case btw and one of Wasm's stated goals), should care:

> As a JS developer, you should care because you have dependencies, and these dependencies have dependencies, you upgrade them, you bundle them. This string stuff is subtle, but also fundamental, and errors are delayed, almost impossible to spot. With the CM, JS developers need a bunch of highly specialized knowledge now that they didn't necessarily need to care about before. I do think that we are acting responsibly when we try our best to let them know about the what and why, so they can address it, either in code or in the spec.

If they are kept in the dark, this will just remain the self-fulfilling prophecy it already is: Wasm does its thing, while all attempts to inform or include JS devs are stonewalled. Greetings to the same Wasm folks up- and downvoting here btw. :)

Reply:
If wasm makes decisions that make JS interop harder *in practice*, that hurts wasm devs, not JS devs, because that will just inhibit use of wasm. I don't see what JS devs can do about it - nor do I think that the entry to the wasm funnel will be on this particular TC39 proposal.

Reply:
That doesn't sound like a good outcome, rather like an argument to better have an FAQ entry so nobody is hurt unknowingly. And for sure this proposal is the entry point: Want to know what the problem is? Direct here. Ran into an issue? Direct here. Component Model? Soon directs here. In a sense, JS and Wasm meet exactly here, where the problem worked around.